File: arrays/Maximum_Product_Difference_Between_Two_Pairs.php
//Not many users of php at leetcode. Pointless to mention the performance. 
class Solution {

    /**
     * @param Integer[] $nums
     * @return Integer
     */
    function maxProductDifference($nums) {
        $length = count($nums);
        sort($nums);
        
        $firstBig = end($nums);
        $secondBig = prev($nums);
        $secondLast = reset($nums);
        $firstLast = next($nums);

        return $firstBig*$secondBig - $firstLast*$secondLast;
    }
}

File: arrays/Maximum_Product_Difference_Between_Two_Pairs.swift
class Solution {
    func maxProductDifference(_ nums: [Int]) -> Int {
        let sorted = nums.sorted();
        print(sorted);

        let len = nums.count;
        let firstBig = sorted[len-1];
        let secondBig = sorted[len-2];
        let firstSmall = sorted[0];
        let secondSmall = sorted[1];
        return firstBig*secondBig-firstSmall*secondSmall;
    }
}
